sqlconnect/internal/snowflake/config.go

```diff
@@ -34,44 +35,80 @@ type Config struct {
 	KeepSessionAlive  bool   `json:"keepSessionAlive"`
 	UseLegacyMappings bool   `json:"useLegacyMappings"`
 	QueryTag          string `json:"queryTag"`
+	Host              string `json:"host"`
+	UseOAuth          bool   `json:"use_oauth"`
+	OAuthToken        string `json:"oauth_token"`
 }
 
 func (c Config) ConnectionString() (dsn string, err error) {
-	sc := gosnowflake.Config{
-		Authenticator: gosnowflake.AuthTypeSnowflake,
-		User:          c.User,
-		Password:      c.Password,
-		Account:       c.Account,
-		Database:      c.DBName,
-		Warehouse:     c.Warehouse,
-		Schema:        c.Schema,
-		Role:          c.Role,
-		Application:   c.Application,
-		LoginTimeout:  c.LoginTimeout,
-		Params:        make(map[string]*string),
-	}
-
-	if c.UseKeyPairAuth {
-		sc.Authenticator = gosnowflake.AuthTypeJwt
-		privateKey, err := c.ParsePrivateKey()
+	if c.UseOAuth {
+		fmt.Println("sqlconnect: Account: " + c.Account)
+		fmt.Println("sqlconnect: Region: " + c.Region)
+		fmt.Println("sqlconnect: Token: " + c.OAuthToken)
+		fmt.Println("sqlconnect: Warehouse: " + c.Warehouse)
+		fmt.Println("sqlconnect: Schema: " + c.Schema)
+		fmt.Println("sqlconnect: Host: " + c.Host)
+		fmt.Println("sqlconnect: DBName: " + c.DBName)
+
+		sc := gosnowflake.Config{
+			Authenticator:    gosnowflake.AuthTypeOAuth,
+			Account:          c.Account,
+			Region:           c.Region,
+			Token:            c.OAuthToken,
+			Warehouse:        c.Warehouse,
+			Schema:           c.Schema,
+			Database:         c.DBName,
+			Host:             c.Host,
+			Protocol:         "https",
+			Port:             443,
```

Comment:
`Host`, `Protocol` & `Port` are not specific to OAuth, are they? Hard coding protocol & port should be avoided too